MyApp/app/controllers/stories_controller.rb

```diff
@@ -0,0 +1,54 @@
+class StoriesController < ApplicationController
+  include Pundit
+
+  before_action :authenticate_user!
+  before_action :set_story, only: [:show, :edit, :update, :destroy]
+  
+  def set_story
+    @story = Story.find(params[:id])
+    authorize @story
+  end
+
+  def index
+    if current_user.admin? || current_user.editor?
+      @stories = Story.most_recent
+    else
+      @stories = policy_scope(Story.most_recent)
+      @stories = @stories.where(user_id: current_user.id)
+    end
```

Comment:
See again how policy_scope works https://github.com/varvet/pundit

I think you should just do:
@stories = policy_scope(Story@stories = policy_scope(Story),
and override Scope subclass in StoryPolicy, then implement logic to decide which scope to use depending on users role